src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -300,12 +305,16 @@ private boolean checkIfNotEmpty(String... values) {
         return Arrays.asList(result);
     }
 
-    private boolean allElementHandled() {
-        return this.jobHandled && this.genericInformationHandled && this.variablesHandled && this.descriptionHandled;
-    }
+    @Override
+    public String getIconPath(List<KeyValueLabelMetadataEntity> keyValueMetadataEntities) {
+        return keyValueMetadataEntities.stream()
+                                       .filter(keyValue -> keyValue.getLabel()
+                                                                   .equals(ATTRIBUTE_GENERIC_INFORMATION_LABEL) &&
+                                                           keyValue.getKey().equals("workflow.icon"))
+                                       .map(keyValue -> keyValue.getValue())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)